packages/application-extension/src/index.tsx

```diff
@@ -415,11 +415,11 @@ const sidebar: JupyterFrontEndPlugin<void> = {
           } else {
             side = 'left';
           }
-        } else if (document.body.dataset.leftSidebarWidget) {
-          id = document.body.dataset.leftSidebarWidget;
+        } else if (app.shell.dataset.leftSidebarWidget) {
```

Comment:
I have no objection to removing `left-sidebar-widget` and `right-sidebar-widget` from the document body, but then we should remove these checks and exit this command early if no sidebar widget is found.

Reply:
Looks good.